0010-zkapp-cli-lightnet-management.md

```diff
@@ -0,0 +1,100 @@
+# RFC for zkapp-cli enhancements to provide the local lightnet networks management
+
+Introduce new sub-commands to the existing zkapp CLI application to streamline the local lightnet networks management for development and testing purposes.
+
+## Summary
+
+We aim to extend the current zkapp CLI application by introducing the `lightnet` sub-command suite. This suite will centralize and simplify Docker containers and provided tools management tasks for local zkApps development and testing, enhancing user and developer experience.
+
+## Motivation
+
+The primary motivation is to simplify the local zkApps development and testing process by offering intuitive CLI commands.
+
+Expected Outcomes:
+
+- A streamlined workflow for managing local lightnet network [Docker containers](https://hub.docker.com/r/o1labs/mina-local-network) and tools provided by such containers.
+- Reduced complexity and better DX, eliminating manual 3rd party tools management.
+
+## Detailed design
+
+### zkapp CLI Interface Enhancements
+
+Under the primary command `zk`, the new `lightnet` sub-commands will be introduced:
+
+#### Help
+
+- **Command**: `zk lightnet`
+- **Output**: Help message with description of available sub-commands and their options.
+
+#### Start lightnet network
+
+- **Command**: `zk lightnet start <options>`
+- **Function**: Checks the env., starts the local lightnet Docker container with reasonable default configuration options and waits until the network sync. Repeating the command while network is up and running should not start a new container, but rather check the status of the existing one.
+- **Configuration options**:
+  - --mode: `single-node` or `multi-node` (default: `single-node`)
+  - --proof-level: `none` or `full` (default: `none`)
+  - --archive-node: `true` or `false` (default: `true`)
+  - --mina-branch: `rampup`, `berkeley` or `develop` (default: `rampup`)
+  - --dune-profile: `lightnet` or `devnet` (default: `lightnet`)
+- **Output**:
+  - Anticipated network properties (as per [Docker image description](https://hub.docker.com/r/o1labs/mina-local-network));
+  - Mina Daemon GraphQL endpoint for communication: [http://localhost:8080/graphql](http://localhost:8080/graphql)
+  - PostgreSQL connection string: `postgresql://postgres:postgres@localhost:5432/archive`;
+  - Mina Daemon and Archive Node logs path;
+  - [o1js configuration](https://github.com/o1-labs/o1js/blob/ccf50e0b58190d9700c6dda4fafee3e62e270131/src/examples/zkapps/hello_world/run_live.ts#L18) code snippet.
+
+#### Stop lightnet network
+
+- **Command**: `zk lightnet stop`
```

Comment:
From my understanding, docker containers will leave behind logs on the user's system. This could introduce a ton of bloat where logs are being kept from each run. Do you think we should wipe the logs out when tearing down the network or have some subcommand to do some house cleaning? Or should we just let the users worry about that, wdyt?

Reply:
Thanks @MartinMinkov,

> docker containers will leave behind logs on the user's system
Which logs you're referring here to?

As it is stated in description to `stop` command:
> In future we might want to extend this command with additional options to preserve the network state between containers lifecycle.
Which essentially means that for this initial version we will wipe everything (containers will start with `--rm` option and we will clean up other leftovers is any). RFC is updated to clearly state about the cleaning up procedure.